hardware_interface/test/test_hardware_components/test_system_with_command_modes.cpp

```diff
@@ -142,8 +142,8 @@ class TestSystemCommandModes : public
     }
 
     // Stopping interfaces
-    for (std::string key : stop_interfaces) {
-      for (uint i = 0; i < info_.joints.size(); i++) {
+    for (const auto & key : stop_interfaces) {
+      for (auto i = 0u; i < info_.joints.size(); i++) {
```

Comment:
would probably be better to use:

`size_t i = 0;`   this is something you change the code before.

Reply:
I think `auto` is generally the better call in almost all occurrences.

Reply:
auto is super nice almost always but at the same time I also like representation-agnostic numeric literals so it's a bit of a bummer

Reply:
Just asked, because Bence commented on my PR not to use `auto i = 0u`. I am OK with any solution, but will try to unify this across the framework.